### ISSUE TITLE
[BUG] output error with compute tally

### ISSUE THREAD

**Issue by ztz1995** [lammps/lammps#2496]
Hello, lammps developers! I'm using lammps-29Oct20. Here is the bug:
The "in.stress" file in example/USER/tally was modified to output some compute tally result to other file. Howerver, there is something wrong with the screen output:
```
Per MPI rank memory allocation (min/avg/max) = 11.31 | 11.64 | 11.71 Mbytes
Step Temp Press c_one[1] c_one[4] 
       0            0    26550.222 7.977747e+13 -1.1634726e+09 
      10    97.993709   -10942.121 -1.0179876e+09 -1.0179876e+09 
      20    114.03944   -36550.435 7.0588548e+13 -1.0294616e+09 
      30    128.48926    6926.5532 -1.1954907e+09 -1.1954907e+09 
      40    140.69154    22117.012 8.7416959e+13 -1.2748867e+09 
      50    143.40291    -21531.67 -1.3252576e+09 -1.3252576e+09 
      60    145.09967   -20346.811 9.1175162e+13 -1.3296962e+09 
      70    154.75218    22500.336 -1.3582148e+09 -1.3582148e+09 
      80    161.99842    9118.1191 9.3850724e+13 -1.3687165e+09 
      90    159.77773   -28190.009 -1.3758816e+09 -1.3758816e+09 
     100    163.93424    -2654.304 9.5397265e+13 -1.3912713e+09 
Loop time of 0.548549 on 24 procs for 100 steps with 4500 atoms
```
where c_one[1] and c_one[4] should be the same, but they are different when fix print is called. The result in the out.txt seems to be right.

The in.stress file is like this:
```
units		real
atom_style	full

read_data	data.spce

pair_style  	lj/cut/coul/long 12.0 12.0
kspace_style	pppm 1.0e-4

pair_coeff	1 1 0.15535 3.166
pair_coeff	* 2 0.0000 0.0000	

bond_style	harmonic
angle_style	harmonic
dihedral_style	none
improper_style	none

bond_coeff	1 1000.00 1.000
angle_coeff	1 100.0 109.47

special_bonds   lj/coul 0.0 0.0 1.0

neighbor        2.0 bin

group		one molecule 1 2

# compute per atom stress contributions
compute		spa all stress/atom NULL pair
# collect stress contributions from all with all
compute		c1 all stress/tally all
# reduce per-atom vector to scalr
compute		one all reduce sum c_c1[1] c_c1[2] c_c1[3] c_spa[1] c_spa[2] c_spa[3]

variable	p1 equal c_one[1]
variable 	p2 equal c_one[4]

fix		2 all nvt temp 300.0 300.0 100.0
timestep	2.0

thermo_style    custom step temp press c_one[1] c_one[4]
thermo		10

fix		printc all print 20 "${p1} ${p2}" file out.txt screen no
run 		100
```
Thanks！

**Comment by akohlmey** [comment]
The computes in the USER-TALLY package use some heuristics to determine whether a computation has been done before or still needs to be done. Apparently, in this case that heuristic didn't work as expected and the computation is repeated leading to bogus numbers the second time. That said, it is generally not a good idea (and can be very detrimental to performance) to trigger the same computation multiple times from different "consumers". A simple way to address this would be to use fix ave/time (or fix ave/atom at the level of per-atom data) as a cache. here is the last part of the provided input changed for using such a cache mechanism:

```
fix  cache   all ave/time 10 1 10 c_one[1] c_one[4]
variable p1 equal f_cache[1]
variable p2 equal f_cache[2]


fix		2 all nvt temp 300.0 300.0 100.0
timestep	2.0

thermo_style    custom step temp press f_cache[1] f_cache[2]
thermo		10

fix		printc all print 20 "${p1} ${p2}" file out.txt screen no
run 		100
```

**Comment by ztz1995** [comment]
@akohlmey 
This method do solve my problem. Thank you!
